### PR TITLE
Use dredd for blueprint testing

### PR DESCRIPTION
**Pull request by kylef**
This pull request uses dredd to start testing the blueprint against the implementation. I had to make a change to the blueprint's vote action to make it return a JSON structure as it should do this.

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -74,11 +74,14 @@ A Question object has the following attributes.
...
++ Response 201 (application/json)
 
-    + Headers
+        {
+            "url": "/questions/1/choices/1",
+            "votes": 1,
+            "choice": "Swift"
+        }
 
-            Location: /questions/1
```
Comment (pksunkara) [inline apiary.apib]: Don't you mean to have both `Body` and `Headers`?

Comment (pksunkara) [inline apiary.apib]: Because, the polls-api server does send `Location` header.

**Comment by pksunkara** [comment]
Minor comment that's blocking. Please address that.

File: apiary.apib (hunk 1 above)
Comment (kylef) [inline apiary.apib]: I've changed the behaviour of the API in 3f902c30390ea476e34f386e232bfb7e039f5619 so it doesn't redirect a Location header to another resource.

**Pull request merged**